### PR TITLE
[WIP] Major revisions: restructure according to in-out philosophy.

### PR DESCRIPTION
**Pull request by rkdarst**
This is a WIP about the revisions.  In particular, the old, non-used slides are still present at the end of the talk.md file, which has to be fixed.  Also there can still be some thing cut down significantly.

My goal was to remove talk of licensing and make it more about the community, what goes in comes out.  There are other sites to talk about licenses once someone is interested.

Also I am not sure about formatting, need to check on cicero and fix things before merging.

===

- Why should people care about licensing, when most people never read
  a license?  It's because if you want to license out, you have to
  license in.  Thus comes the whole idea of open science: what comes
  in goes out, and citations go back.

### CODE REVIEW

File: talk.md [hunk 1]
```diff
@@ -52,37 +73,39 @@ layout: false
 
 Curiosity: Not too many projects consider a software management plan **yet**.
 
-.emph[The big keyword in this talk is **reproducibility**.]
+Data/software should make science completely **shareable** and
+**reproducible**... .emph[but is it?
```
Comment (bast) [inline talk.md]: you need to close the `emph`

**Comment by bast** [comment]
After a quick scroll-through I like the changes. Some slides can definitely be dropped. But I think that a discussion about fears of sharing and the discussion of what is and what is not derivative work is important (from the diff it was not clear to me whether the plan is to drop these). 

If you want to restructure significantly (which is totally fine) at late stage (which is fine, I fully understand), I perhaps recommend to create a branch or a new repo and later we combine?

I think the exercises are great but I think a discussion about derivative work and fears of sharing is more important than an exercise about trademarks.

Again, you have full flexibility if we branch off or spin-off a new repo.

**Comment by bast** [comment]
If these two aspects remain somehow (does not have to be in previous form) in the core part of the talk, then I am happy to do this on master but derivative work is a core concern for any researcher IMO.

**Comment by bast** [comment]
Don't worry too much about formatting. Simply hash out the bullet points and I can fix formatting tonight. Also I think trademarks are not a problem for most researchers. I think it is OK to mention them but I would dedicate exercise time on copyright and not on patents or trademarks.

**Comment by rkdarst** [comment]
Thanks for the feedback.

I agree with derivative work parts.  I have an idea: remove the reproducibility slides, and put longer derivative work discussion there.  It really is central to the message: what you take in, you make a derivative work of, then it goes out.  So it needs top emphasis.  This will save time later and makes things more clear.

The patent/trademark is designed to be a fast discussion, mainly to contrast copyright and show that copyright isn't the only thing or the right tool to solve all problems.  But you are right there are too many exercises on that, I will try to remove some.

We can leave this as separate branch and merge later if needed, no need to decide now.

**Comment by bast** [comment]
For my reference these are the new slides (makes it a bit easier for me to review rather than looking at displaced diff): https://cicero.xyz/v3/remark/0.14.0/github.com/rkdarst/social-coding/rkd-revisions/talk.md

**Comment by bast** [comment]
I suggest you serve it from the branch, then you are in full control and we can later merge the changes to master. Please update the author on first slide, please also later update the link here: https://coderefinery.org/workshops/2018-12-11-espoo/

If you prefer, you can also create a branch on the central repo and call it `2018-12-11-espoo` then you don't have to keep the fork and branch forever to not break links.

If you prefer to merge to master later tonight, we can probably also manage that.

**Comment by rkdarst** [comment]
Yeah, own branch is good.  Can you allow me to make a branch on this repo?

**Comment by rkdarst** [comment]
To do before merge:
- [ ] "social science" -> "social coding"
- [ ] there are some duplicate slides in there
- [ ] as usual, shorter throughout

Also, I think that the cake analogy could use some ordering and/or updating. At least, when I was presenting, it would have made more sense to start with the most open license first (viral GPL) and discuss how this is the most similar to expectations of academic code ("open if use is similar").  Then discuss more permissive.  But overall, in a talk designed to be short and not mainly about licensing, I'm not sure how well the cake analogy still fits.  Thoughts?

**Comment by bast** [comment]
I agree. There are too many slides. If we have 1 h time and want to have a discussion, we should have maximum 20-25 slides.

I would like to not delete the cakes but spin-off into a blog post or something.

**Comment by bast** [comment]
We also need more tangible examples (we should spin this off into an issue before merging): Today while listening as participant I would have thought: "Hmm, yes, I understand why these examples were a problem for Apple or Debian or Firefox, but I am not sure I will ever have one of these problems." - we need examples to which almost everybody can relate to.

**Comment by rkdarst** [comment]
What do you think of the new idea of beginning the presentation, the "in and out" strategy?

- [ ] Remove trademarks, in some of the places put "data" there instead
- [ ] some intro slides are redundant with the new way of presenting
- [ ] remove all current examples.  replace them with actual or hypotethical examples from code
- ]

**Comment by bast** [comment]
I like it - in particular the very good observation and statement that if one is not careful about licenses/copyright when mixing other people's work (in), it can become difficult or impossible to share your own work (out) or even use your own work in future. It never occurred to me in such a clear way. This is a strong message and we need to make sure everybody got it.

**Comment by rkdarst** [comment]
- [ ] create an open core (open) + your value on top (initially closed).  Separate secrets from repo.

**Comment by bast** [comment]
Superseded by #8. Closing this one.

**Pull request closed without merging**